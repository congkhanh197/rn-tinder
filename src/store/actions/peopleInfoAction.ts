import * as C from '../constants';

const getPersonInfoAction = () => ({
  type: C.GET_PERSON_INFO_REQUEST,
});

export {getPersonInfoAction};
