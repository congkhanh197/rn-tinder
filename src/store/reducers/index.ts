import { combineReducers } from "redux";
import peopleInfoReducer, {IPeopleResponse} from "./peopleInfoReducer";
import favoriteReducer from "./favoriteReducer";
export default combineReducers({ peopleInfoReducer, favoriteReducer });
