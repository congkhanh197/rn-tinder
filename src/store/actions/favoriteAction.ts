import * as C from '../constants';
import {IPeopleResponse} from '../reducers/peopleInfoReducer';
export const addFavoritePeopleAction = (payload: IPeopleResponse) => ({
  type: C.ADD_PERSON_FAVORITE,
  payload,
});
export const retrieveFavoritePeopleDataAction = () => ({
  type: C.RETRIEVE_FAVORITE_PEOPLE_DATA,
});
