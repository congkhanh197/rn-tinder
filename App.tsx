import React, {useState, useEffect} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  View,
  Text,
  StatusBar,
  Platform,
} from 'react-native';

import {
  Header,
  LearnMoreLinks,
  Colors,
  DebugInstructions,
  ReloadInstructions,
} from 'react-native/Libraries/NewAppScreen';

import {Provider} from 'react-redux';
import store from './src/store';
import AppNavigator from './src/navigation/AppNavigator';
import {retrieveFavoritePeopleDataAction} from './src/store/actions/favoriteAction';

// import Icon from 'react-native-vector-icons/FontAwesome';
// import Ionicons from 'react-native-vector-icons/Ionicons';
// const myIcon = <Icon name="rocket" size={30} color="#900" />;

const App = () => {
  useEffect(() => {
    store.dispatch(retrieveFavoritePeopleDataAction());
  }, []);
  return (
    <Provider store={store}>
      <StatusBar barStyle="dark-content" />
      <SafeAreaView>
        {Platform.OS === 'ios' && <StatusBar barStyle="default" />}
        <AppNavigator />
      </SafeAreaView>
    </Provider>
  );
};

const styles = StyleSheet.create({
  scrollView: {
    backgroundColor: Colors.lighter,
  },
  engine: {
    position: 'absolute',
    right: 0,
  },
  body: {
    backgroundColor: Colors.white,
  },
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
    color: Colors.black,
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
    color: Colors.dark,
  },
  highlight: {
    fontWeight: '700',
  },
  footer: {
    color: Colors.dark,
    fontSize: 12,
    fontWeight: '600',
    padding: 4,
    paddingRight: 12,
    textAlign: 'right',
  },
});

export default App;
