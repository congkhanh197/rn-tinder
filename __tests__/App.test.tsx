import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import store from '../src/store';
import { retrieveFavoritePeopleDataAction } from '../src/store/actions/favoriteAction';

jest
  .mock('../src/store', () => ({
    dispatch: jest.fn(),
    subscribe: jest.fn(),
    getState: jest.fn(),
  }))
  .mock('../src/store/actions/favoriteAction', () => ({
    retrieveFavoritePeopleDataAction: jest.fn(()=>({
      type: "RETRIEVE_FAVORITE_PEOPLE_DATA",
    })),
  }))
  .mock('../src/navigation/AppNavigator', () => () => 'Test');
describe.only('Test render app', () => {
  test('Check the render method of app', () => {
    const tree = renderer.create(<App />);
    expect(tree).toMatchSnapshot();
  });
  test('Should call useEffect', () => {
    const wrapper = mount(<App />);
    expect(store.dispatch).toHaveBeenCalledWith(retrieveFavoritePeopleDataAction())
  });
});
