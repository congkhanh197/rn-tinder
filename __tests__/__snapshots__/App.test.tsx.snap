// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render app Check the render method of app 1`] = `"Test"`;
